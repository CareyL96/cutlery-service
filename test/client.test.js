
const { shallow, render } = require('enzyme');
const axios = require('axios');
const React = require('react');
import renderer from 'react-test-renderer';

const TestData = [
  {
    "id": 52,
    "name": "Fantastic Soft Table",
    "description": "Ea tempore rerum assumenda et deserunt consequatur nemo eos. Omnis temporibus voluptas. Eaque veniam laboriosam iusto maxime aut ipsam voluptatum. Ad officiis cum quia hic numquam assumenda unde et. Quo cumque consequatur dolorem distinctio in rerum in at. Voluptatem quia eius velit.",
    "manufacturer": "Kling, Towne and Waelchi",
    "price": "496.00",
    "imageurls": [
        "https://s3-us-west-1.amazonaws.com/kitchen-photos/images/image-1.jpg",
        "https://s3-us-west-1.amazonaws.com/kitchen-photos/images/image-16a.jpg",
        "https://s3-us-west-1.amazonaws.com/kitchen-photos/images/image-1b.jpg"
    ]
  },
  {
    "id": 51,
    "name": "Awesome Soft Computer",
    "description": "Ut officia et nisi maiores possimus hic nostrum. Vel et eos consequatur. Molestiae accusamus rerum eveniet dolores modi reprehenderit.",
    "manufacturer": "Konopelski - Satterfield",
    "price": "458.00",
    "imageurls": [
        "https://s3-us-west-1.amazonaws.com/kitchen-photos/images/image-18.jpg"
    ]
  }
]
// test App
const { App } = require('../client/components/App.jsx')
describe('App.jsx', () => {
  //check on call, grid was rendered
  test('Grid rendered by App', () => {
    const shallowWrapper = shallow(< App />);
    expect(shallowWrapper.find('Grid').length).toBe(1);
    });

  test('App makes get request on mount and updates state with result', () => {
    const spyGet = jest.spyOn(axios, 'get');
    const shallowWrapper = shallow(< App />);
    expect(spyGet).toHaveBeenCalled();
  });
  test('Calling change tool will set state', () => {
    const shallowWrapper = shallow(< App />);
    const spyState = jest.spyOn(shallowWrapper.instance(), 'setState');
    shallowWrapper.instance().changeTool(1);
    expect(spyState).toHaveBeenCalled();
  });
})

const { Detail } = require('../client/components/Detail.jsx');
describe('Detail.jsx', () => {
  test('Detail component and children match snapshot', () => {
  const component = renderer.create(<Detail tool={TestData[0]} />);
  let jsonComponent = component.toJSON();
  expect(jsonComponent).toMatchSnapshot();
  });
});


const { List } = require('../client/components/List.jsx');
describe('List.jsx', () => {
  const spyGet = jest.spyOn(axios, 'get');
  spyGet.mockResolvedValue(TestData);
  const wrapper = shallow(< App />);
  test('List should contain as many items as there are tools', () => {
    expect(wrapper.find(List).length).toBe(1);
    const listWrapper = wrapper.find(List).dive();
    console.dir(listWrapper.find('SingleTool'));
    expect(listWrapper.find('SingleTool').length).toBe(TestData.length);
  });

})

