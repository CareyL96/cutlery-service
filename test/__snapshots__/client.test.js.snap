// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail.jsx Detail component and children match snapshot 1`] = `
<div
  className="container-fluid"
>
  <div
    className="show-grid row"
  >
    <div
      className="col-md-6 col-sm-6"
    >
      <div
        className="container-fluid"
      >
        <div
          className="show-grid row"
        >
          <div
            className="no-gutter col-md-12"
          >
            <img
              className="img-responsive"
              src="https://s3-us-west-1.amazonaws.com/kitchen-photos/images/image-1.jpg"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="no-gutter col-md-3 col-sm-6 col-xs-6"
          >
            <img
              className="img-responsive"
              onClick={[Function]}
              src="https://s3-us-west-1.amazonaws.com/kitchen-photos/images/image-1.jpg"
            />
          </div>
          <div
            className="no-gutter col-md-3 col-sm-6 col-xs-6"
          >
            <img
              className="img-responsive"
              onClick={[Function]}
              src="https://s3-us-west-1.amazonaws.com/kitchen-photos/images/image-16a.jpg"
            />
          </div>
          <div
            className="no-gutter col-md-3 col-sm-6 col-xs-6"
          >
            <img
              className="img-responsive"
              onClick={[Function]}
              src="https://s3-us-west-1.amazonaws.com/kitchen-photos/images/image-1b.jpg"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-md-6 col-sm-6"
    >
      <div
        className="container-fluid"
      >
        <div
          className="row"
        >
          <div
            className="col-md-12"
          >
            <h3>
              Fantastic Soft Table
            </h3>
             by 
            <h4>
              Kling, Towne and Waelchi
            </h4>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-12"
          >
            <p>
               
              Ea tempore rerum assumenda et deserunt consequatur nemo eos. Omnis temporibus voluptas. Eaque veniam laboriosam iusto maxime aut ipsam voluptatum. Ad officiis cum quia hic numquam assumenda unde et. Quo cumque consequatur dolorem distinctio in rerum in at. Voluptatem quia eius velit.
               
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-1 col-md-push-7"
    >
      $
      496.00
    </div>
    <div
      className="col-md-4 col-md-push-7"
    >
      <button
        className="btn btn-lg btn-default"
        disabled={false}
        type="button"
      >
        Add To Basket
      </button>
    </div>
  </div>
</div>
`;
